lib/dry/initializer/builders/reader.rb

```diff
@@ -36,9 +36,10 @@ def attribute_line
     def method_lines
       return unless @reader
       return unless @null
+
       [
         "def #{@target}",
-        "  #{@ivar} unless #{@ivar} == Dry::Initializer::UNDEFINED",
+        "  #{@ivar} unless #{@ivar} == Dry::Initializer::UNDEFINED && !#{@null == true}",
```

Comment:
There's no need to make the check at the runtime.
It would be better to provide `@null`-dependent versions of the reader